app/src/main/java/ru/thstdio/aa2020/ui/list/MoviesListViewModel.kt

```diff
@@ -1,23 +1,63 @@
 package ru.thstdio.aa2020.ui.list
 
+import android.util.Log
 import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import androidx.paging.PagedList
-import androidx.paging.toLiveData
-import ru.thstdio.aa2020.api.Repository
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import ru.thstdio.aa2020.cache.ITEMS_SIZE_IN_PAGE
+import ru.thstdio.aa2020.cache.Repository
 import ru.thstdio.aa2020.data.Cinema
 
 
 class MoviesListViewModel(private val repository: Repository) :
     ViewModel() {
-    val pagedListLiveData: LiveData<PagedList<Cinema>> = createPagedListLiveData()
-
-    private fun createPagedListLiveData(): LiveData<PagedList<Cinema>> {
-        val dataSourceFactory: CinemaDataSourceFactory = CinemaDataSourceFactory(
-            repository,
-            viewModelScope
-        )
-        return dataSourceFactory.toLiveData(pageSize = 20)
+
+
+    private var loadNewPageJob: Job? = null
+    private val _cinemas: MutableLiveData<List<Cinema>> = MutableLiveData()
+    val cinemas: LiveData<List<Cinema>> get() = _cinemas
+    private val _scrollListenerStatus: MutableLiveData<Boolean> = MutableLiveData()
+    val scrollListenerStatus: LiveData<Boolean>
+        get() = _scrollListenerStatus
+
+    private var currentPage = 1
+    private val exceptionHandler = CoroutineExceptionHandler { coroutineContext, exception ->
+        println("CoroutineExceptionHandler got $exception in $coroutineContext")
+    }
+
+    init {
+        loadNewPageJob = viewModelScope.launch(exceptionHandler) {
+            _scrollListenerStatus.value = false
```

Comment:
u can remove _scrollListenerStatus.value = false here and set default falue as false in _scrollListenerStatus